e2e-pw/src/oss/specs/smoke-tests/embeddings.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { test as base } from "src/oss/fixtures";
+import { GridPom } from "src/oss/poms/grid";
+import { PanelPom } from "src/oss/poms/panels/panel";
+import { EmbeddingsPom } from "src/oss/poms/panels/embeddings-panel";
+import { getUniqueDatasetNameWithPrefix } from "src/oss/utils";
+
+const datasetName = getUniqueDatasetNameWithPrefix("smoke-quickstart");
+
+const test = base.extend<{ grid: GridPom; panel: PanelPom }>({
+  grid: async ({ page, eventUtils }, use) => {
+    await use(new GridPom(page, eventUtils));
+  },
+  panel: async ({ page }, use) => {
+    await use(new PanelPom(page));
+  },
+  embeddings: async ({ page, eventUtils }, use) => {
+    await use(new EmbeddingsPom(page, eventUtils));
+  },
+});
+
+test.beforeAll(async ({ fiftyoneLoader }) => {
+  await fiftyoneLoader.executePythonCode(
+    `
+        import fiftyone as fo
+        import fiftyone.zoo as foz
+        import fiftyone.brain as fob
+        import numpy as np
+
+        dataset = foz.load_zoo_dataset("quickstart", max_samples=5, dataset_name="${datasetName}")
+        dataset.persistent = True
+
+        embeddings = np.random.random((5, 512))
+        fob.compute_visualization(dataset, brain_key="img_viz", embeddings=embeddings)
+
+        dataset.save()
+    `
+  );
+});
+
+test.beforeEach(async ({ page, fiftyoneLoader }) => {
+  await fiftyoneLoader.waitUntilGridVisible(page, datasetName);
+});
+
+test.describe("quickstart", () => {
+  test("embeddings panel opens", async ({ panel, embeddings }) => {
+    await panel.open("Embeddings");
+    await embeddings.asserter.verifySelectorVisible();
+  });
+
+  test("lasso samples work", async ({ panel, embeddings }) => {
+    await panel.open("Embeddings");
+    await embeddings.asserter.verifyLassoSelectsSamples();
+  });
+});
```

Comment:
I think this will have to be combined into one test until routing changes are merged. The space layout is persisting because of the session, i.e. the embeddings panel is already open in the second test so Line 51 fails

Reply:
cool! I have had this issue before :) The single tests running alone passes but a combination of tests fail because the assumption of each test starts in a clean slate (here meaning embeddings panel close). I can fix this by "cleaning" the changes after each test.